src/rgw/rgw_rest_client.cc

```diff
@@ -454,7 +454,7 @@ int RGWRESTSimpleRequest::forward_request(const DoutPrefixProvider *dpp, const R
 
   int r = process(dpp, y);
   if (r < 0){
-    if (r == -EINVAL){
+    if (r == -EINVAL && http_status == 0) { // make sure no header were received
```

Comment:
I wonder if we can just leave `http_status` in the if clause so we return 503 on all internal curl errors no matter what the return code is.

Reply:
should probably be ok. cc @cbodley

Reply:
Updated with only the http_status case.